lib/plugins/healthcheckMetricsPlugin.ts

```diff
@@ -0,0 +1,83 @@
+import type { Either } from '@lokalise/node-core'
+import type { FastifyInstance } from 'fastify'
+import fp from 'fastify-plugin'
+
+const VALID_PROMETHEUS_NAME_REGEX = /[a-zA-Z_:][a-zA-Z0-9_:]*/
+
+export interface HealthcheckMetricsPluginOptions {
+  healthChecks: readonly PrometheusHealthCheck[]
+}
+
+export type HealthcheckResult = {
+  checkTimeInMsecs: number
+  checkPassed: boolean
+}
+
+export type PrometheusHealthCheck = {
```

Comment:
open to suggestions of a better name! `HealthCheck` is already taken by publicHealthcheckPlugin